### ISSUE TITLE
TLS server not accepting connections

### ISSUE THREAD

**Issue by jchavanton** [pjsip/pjproject#2380]
**Describe the bug**
Not accepting TLS connections anymore.

**To Reproduce**

Start a server :
```
17:14:52.420             tlstp:5056  SIP TLS listener is ready for incoming connections at 1.75.96.70:5056
[17:14:52.420][INFO] main: TLS tcfg.tlsConfig.ca_list      :/tls/ca_list.pem
[17:14:52.420][INFO] main: TLS tcfg.tlsConfig.certFile     :/tls/cert.pem
[17:14:52.420][INFO] main: TLS tcfg.tlsConfig.privKeyFile  :/tls/key.pem
[17:14:52.420][INFO] main: TLS tcfg.tlsConfig.verifyServer :1
[17:14:52.420][INFO] main: TLS tcfg.tlsConfig.verifyClient :0
```

Make a call, the TLS connection will connect but fail ...

```
:30.552      ssl0x7f3c50000b20  Handshake failed in accepting 1.75.69.51:58997: End of file (PJ_EEOF)
```

If I checkout an older commit before the refactoring,  it works :

git checkout b4a85b0509dc45eb2f642b7bbaf16a9a7d398386

Seems like there was a lot of refactoring, I think this should be easy to reproduce ?

git diff master..b4a85b0509dc45eb2f642b7bbaf16a9a7d398386 pjsip/src/pjsip/sip_transport_tls.c

**Comment by jchavanton** [comment]
I went to the painful path of testing by commit isolation

reverting this commit is fixing the problem for me
```
commit a190e94028ceb860472a03e8145298dce26706c3
Author: Sauw Ming <ming@teluu.com>
Date:   Fri Aug 23 04:53:05 2019 +0000

    Fixed #2220: Add list of trusted root CA for OpenSSL
```

**Comment by jchavanton** [comment]
Saving some test notes, in case something was wrong.

```
// test 4 : not working
commit 3e7b75cb2e482baee58c1991bd2fa4fb06774e0d (tag: 2.10)
Merge: e5943ba8e 6ea8618d9
Author: mingteluu <ming@teluu.com>
Date:   Fri Feb 14 16:48:27 2020 +0700

// test 5 : not working
commit 81b29e9aa21bc9ae056801f8b3fad49496167df8
Author: Sauw Ming <ming@teluu.com>
Date:   Fri Nov 29 07:53:43 2019 +0000

// test 7 : not working 
commit 8a15cf5dceb5f6bda8c47816be1d191513d3be92 (HEAD)
Author: Riza Sulistyo <riza@teluu.com>
Date:   Wed Aug 28 09:32:26 2019 +0000

// test 10 : not working 
commit a190e94028ceb860472a03e8145298dce26706c3
Author: Sauw Ming <ming@teluu.com>
Date:   Fri Aug 23 04:53:05 2019 +0000

    Fixed #2220: Add list of trusted root CA for OpenSSL
    
    
    >>> a190e94028ceb860472a03e8145298dce26706c3 <<<

// test 11 : (reverting) working 
commit a190e94028ceb860472a03e8145298dce26706c3
Author: Sauw Ming <ming@teluu.com>
Date:   Fri Aug 23 04:53:05 2019 +0000

    Fixed #2220: Add list of trusted root CA for OpenSSL



// test 9 : working
commit 3464650ee00d5e8b8fd4d17b7b1906b6b8a5fe3e
Author: Nanang Izzuddin <nanang@teluu.com>
Date:   Mon Jul 29 02:32:55 2019 +0000


// test 8 : working
commit 3f58d4d10dfe01b51cca84449c18f0bb6fa6b473
Author: Sauw Ming <ming@teluu.com>
Date:   Mon Jul 1 07:12:43 2019 +0000


// test 6 : working
commit d059927e63ed31fb0008e2fb9c1d4ad67a5828bc
Author: Sauw Ming <ming@teluu.com>
Date:   Mon May 27 03:54:27 2019 +0000

    Fixed #2203: Failure to set minimum SSL version when OpenSSL 1.1 is used


// test 3 : working
commit 765e804fdad726f71112b742bcc4d16bacd8b468
Author: Riza Sulistyo <riza@teluu.com>
Date:   Fri May 24 03:32:17 2019 +0000

// test 2 : working 
26d59020b3ebd87b4fa747733fddc3ac02ad13e1






-----

// testing 
./configure --disable-libwebrtc && make dep && make 
make install && cd .. && make clean && make
/git/voip_patrol/voip_patrol -p 5055 -c /voip_patrol/xml/customer_server.xml --tls-calist "/tls/ca_list.pem" --tls-privkey "/tls/key.pem" --tls-cert "/tls/cert.pem" --tls-verify-server --debug-console 10
```

**Comment by sauwming** [comment]
Since the issue has been quite a while and may no longer be valid, let us know if it still persists in the latest version.
